src/typings/targetDescriptor.types.ts

```diff
@@ -36,10 +36,20 @@ export interface LineNumberPosition {
   lineNumber: number;
 }
 
-export interface LineNumberMark {
-  type: "lineNumber";
+export interface LineNumberRange {
+  rangeType: RangeType;
   anchor: LineNumberPosition;
   active: LineNumberPosition;
+  excludeAnchor: boolean;
+  excludeActive: boolean;
+}
+
+export interface LineNumberMark {
+  type: "lineNumber";
+  // Either (anchor AND active) OR elements must be set.
+  anchor?: LineNumberPosition;
+  active?: LineNumberPosition;
+  elements?: LineNumberRange[];
```

Comment:
I think I have changed my mind. I think it's okay that a single mark/word modifier has a range. It still just only one target, but baking all the logic of compound targets into this mark is something I rather avoid.